src/sqlitetablemodel.cpp

```diff
@@ -432,6 +432,26 @@ bool SqliteTableModel::removeRows(int row, int count, const QModelIndex& parent)
     return true;
 }
 
+QModelIndex SqliteTableModel::dittoRecord(int old_row)
+{
+    int firstEditedColumn = 0;
+    int new_row = rowCount() - 1;
+
+    sqlb::Table t = sqlb::Table::parseSQL(m_db->getObjectByName(m_sTable).getsql()).first;
+    for (int col = 0; col < t.fields().size(); ++col) {
+        if (!t.fields().at(col)->primaryKey()) {
+            if (!firstEditedColumn)
+                firstEditedColumn = col + 1;
+
+            QVariant value = m_data.at(old_row).value(col + 1);
+            if (!value.isNull())
```

Comment:
Aah, I see. Good you're mentioning it :smiley: You should be able to get the actual content using a construction similar to this one: ```QVariant data =data(indexrowi, column), Qt::EditRole);``` The important bit being the ```Qt::EditRole```. We're returning 'NULL' for the display role, but nothing for the edit role. Does this help?